### PR TITLE
update flecsi and legion package.py

### PR DESCRIPTION
**Pull request by ktsai7**

### CODE REVIEW

File: var/spack/repos/builtin/packages/legion/package.py [hunk 1]
```diff
@@ -69,7 +72,9 @@ def cmake_args(self):
             '-DBUILD_SHARED_LIBS=%s' % ('+shared' in self.spec)]
 
         if self.spec.variants['build_type'].value == 'Debug':
-            cmake_cxx_flags.append('-DDEBUG_REALM', '-DDEBUG_LEGION', '-ggdb')
+            cmake_cxx_flags.append('-DDEBUG_REALM')
+            cmake_cxx_flags.append('-DDEBUG_LEGION')
+            cmake_cxx_flags.append('-ggdb')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/legion/package.py]: You can use extend to add a list of flags

Comment (ktsai7) [reply var/spack/repos/builtin/packages/legion/package.py]: Like 
```
cmake_cxx_flags.extend('-DDEBUG_REALM')
cmake_cxx_flags.extend('-DDEBUG_LEGION')
cmake_cxx_flags.extend('-ggdb')
```
?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/legion/package.py]: Like:
```python
cmake_cxx_flags.extend([
    '-DDEBUG_REALM',
    '-DDEBUG_LEGION',
    '-ggdb',
])
```

Comment (ktsai7) [reply var/spack/repos/builtin/packages/legion/package.py]: Updated.

**Pull request merged**